azure-pipelines.yml

```diff
@@ -42,18 +52,19 @@ stages:
     - checkout: sonic-mgmt-common
       clean: true
       submodules: recursive
-      displayName: 'Checkout code'
+      displayName: 'Checkout sonic-mgmt-common'
 
     - task: DownloadPipelineArtifact@2
       inputs:
         source: specific
         project: build
-        pipeline: 1
+        pipeline: 142
```

Comment:
You can dynamically retrieve the pipeline ID using the Azure DevOps CLI or REST API. Then, set the pipeline ID as a variable to avoid hardcoding it in your configuration.

Reply:
This is just a declarative yaml file, cannot run commands.. I just followed the existing logic where pipeline id 1 was hardcoded.. There is no dynamic pipeline requirement here -- idea is to download yang & libyang from the latest official vs build, which is [pipeline 142](https://dev.azure.com/mssonic/build/_build?definitionId=142). Anyway tried changing it to the pipeline name and it worked.

Reply:
The Azure pipeline YAML configuration allows you to dynamically retrieve and use the pipeline ID to avoid hardcoding it. 
e.g.

```
- script: |
    PIPELINE_ID=$(curl -k "https://dev.azure.com/mssonic/build/_apis/pipelines?api-version=7.1" | jq -r '.value[] | select(.name=="Azure.sonic-buildimage.official.generic") | .id')
  displayName: 'Get Pipeline ID'

- task: DownloadPipelineArtifact@2
  inputs:
    source: specific
    project: build
    pipeline: $(PIPELINE_ID)
. . .
```

Reply:
Will the script local variable automatically get exported as pipeline variable?

Anyway, did you check the latest? I have used the pipeline name instead of id.

Reply:
Okay, I notice the pipeline id has been replaced with the pipeline name. No hardcoding, which resolves my comment.